index.js

```diff
@@ -356,7 +356,7 @@ module.exports = (command, args, options) => {
 	spawned.then = (onFulfilled, onRejected) => handlePromise().then(onFulfilled, onRejected);
 	spawned.catch = onRejected => handlePromise().catch(onRejected);
 	spawned.cancel = function () {
-		if (!spawned.killed) {
+		if (!spawned.killed && spawned.exitCode !== 0) {
```

Comment:
In which situation can `spawned` be `killed` but with an `exitCode` of `0`?

Reply:
oh sorry! I should check for something else...but that won't work

Reply:
I am not wrong then exitcode is 0 when the process was successful. this would just check whether the proccess hasn't successfully exited and can be no longer killed

Reply:
As far as I understand the only thing we need to check is whether the process has already terminated. It should not matter whether it was successful or not, and whether it was done through a signal (`killed`) or not. 

This is already given by Node.js with the return value of `spawned.kill()` so wouldn't this work instead?

```js
if (canceled) { return }

if (process.kill()) {
  canceled = true;
}
```

Reply:
This would work but then what would be the use of `canceled` at all. We won't be able to use it. On killing the process, processDone promise would get resolved

```
const processDone = new Promise(resolve => {
                // some code
		spawned.on('error', error => {
			cleanup();
			resolve({error});  // because of this
		});
                // more code
	});
```

and then in handlePromise function this condition check would pass with canceled still being `false`.

```
if (result.error || result.code !== 0 || result.signal !== null || canceled) {
			const error = makeError(result, {
				joinedCommand,
				parsed,
				timedOut,
				canceled
			});
```
 Thus, the error made by makeError won't be as desired - having canceled == true and a proper message

Reply:
all what is mean is that we can;t execute `kill()` before making `canceled` `true`

Reply:
Isn't `handlePromise` only resolved on the next tick?  I.e. the order would be: 
  1) `spawned.kill()` is fired
  2) `canceled = true` is executed
  3) `handlePromise` callback is executed

Correct me if I'm wrong.

Reply:
oh! this does work. not sure why but I tried this in morning(kill before making canceled true) and this threw that error.

Reply:
You actually made me double check, I was not sure either whether `exit` would be fired on the next tick or not.